sklearn/compose/_column_transformer.py

```diff
@@ -1394,3 +1490,99 @@ def __call__(self, df):
         if self.pattern is not None:
             cols = cols[cols.str.contains(self.pattern, regex=True)]
         return cols.tolist()
+
+
+class _RemainderColsList(UserList):
+    """A list that raises a warning whenever items are accessed.
+
+    It is used to store the columns handled by the "remainder" entry of
+    ``ColumnTransformer.transformers_``, ie ``transformers_[-1][-1]``.
+
+    For some values of the ``ColumnTransformer`` ``transformers`` parameter,
+    this list of indices will be replaced by either a list of column names or a
+    boolean mask; in those cases we emit a ``FutureWarning`` the first time an
+    element is accessed.
+
+    Parameters
+    ----------
+    columns : list of int
+        The remainder columns.
+
+    future_dtype : {'str', 'bool'}, default=None
+        The dtype that will be used by a ColumnTransformer with the same inputs
+        in a future release. There is a default value because providing a
+        constructor that takes a single argument is a requirement for
+        subclasses of UserList, but we do not use it in practice. It would only
+        be used if a user called methods that return a new list such are
+        copying or concatenating `_RemainderColsList`.
```

Comment:
I don't understand this part. I tried and managed to make a subclass of UserList with a required kwarg.